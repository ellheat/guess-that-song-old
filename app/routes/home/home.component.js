import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import envConfig from 'env-config';

import messages from './home.messages';
import { MaintainerList } from './maintainerList/maintainerList.component';
import { Container, Title, TitleLogo, EnvName } from './home.styles';

export class Home extends PureComponent {
  static propTypes = {
    items: PropTypes.object,
    language: PropTypes.string.isRequired,
    fetchMaintainers: PropTypes.func.isRequired,
    setLanguage: PropTypes.func.isRequired,
    match: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  };

  componentDidMount() {
    this.props.fetchMaintainers(this.props.language);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.language !== this.props.language) {
      this.props.fetchMaintainers(this.props.language);
    }
  }

  render() {
    return (
      <Container>
        <Helmet title="Homepage" />

        <Title>
          <TitleLogo name="logo" />
          <FormattedMessage {...messages.welcome} />
        </Title>

        <EnvName>Environment: {envConfig.name}</EnvName>

        <MaintainerList items={this.props.items} />
      </Container>
    );
  }
}
