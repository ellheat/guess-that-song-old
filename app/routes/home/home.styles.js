import styled from 'styled-components';

import { SpriteImage } from '../../theme';

export const Container = styled.div`
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
 text-align: center;
`;

export const Title = styled.h1`
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
`;

export const TitleLogo = SpriteImage.extend`
  margin-bottom: 16px;
`;

export const EnvName = styled.div`
  color: green;
  margin-top: 40px;
  margin-bottom: 40px;
`;
