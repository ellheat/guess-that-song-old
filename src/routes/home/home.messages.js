import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'home.title',
    defaultMessage: 'Guess that song!',
  },
});
