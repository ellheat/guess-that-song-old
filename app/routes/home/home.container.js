import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { hot } from 'react-hot-loader';
import { withRouter } from 'react-router-dom';
import { compose } from 'ramda';

import { Home } from './home.component';
import { MaintainersActions } from '../../modules/maintainers/maintainers.redux';
import { selectMaintainersItems } from '../../modules/maintainers/maintainers.selectors';
import { LocalesActions } from '../../modules/locales/locales.redux';
import { selectLocalesLanguage } from '../../modules/locales/locales.selectors';


const mapStateToProps = createStructuredSelector({
  items: selectMaintainersItems,
  language: selectLocalesLanguage,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  fetchMaintainers: MaintainersActions.fetch,
  setLanguage: LocalesActions.setLanguage,
}, dispatch);

export default compose(
  hot(module),
  connect(mapStateToProps, mapDispatchToProps),
  withRouter
)(Home);
