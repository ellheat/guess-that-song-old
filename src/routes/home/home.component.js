import React, { PureComponent } from 'react';
import Helmet from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { ROUTES } from '../../constants';
import { colors } from '../../theme/styled';

import messages from './home.messages';
import { Container, Title } from './home.styles';
import { BoxList } from './boxList/boxList.component';

const BOXES = [
  {
    title: 'Jukebox',
    color: colors.blue,
    url: ROUTES.jukebox,
    disabled: true,
  },
  {
    title: 'Multiplayer',
    color: colors.green,
    url: ROUTES.multiplayer,
  },
  {
    title: 'Speed',
    color: colors.yellow,
    url: ROUTES.speed,
    disabled: true,
  },
];

export class Home extends PureComponent {
  render() {
    return (
      <Container>
        <Helmet title="Homepage" />

        <Title>
          <FormattedMessage {...messages.title} />
        </Title>

        <BoxList list={BOXES} />
      </Container>
    );
  }
}
