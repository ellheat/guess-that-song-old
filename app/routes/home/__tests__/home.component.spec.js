import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { fromJS } from 'immutable';

import { Home } from '../home.component';


describe('Home: Component', () => {
  const defaultProps = {
    fetchMaintainers: () => {},
    items: fromJS([{ name: 'name-1' }, { name: 'name-2' }, { name: 'name-3' }]),
    language: 'en',
    setLanguage: () => {},
    location: {},
    match: {},
    history: { push: () => {} },
  };

  const component = (props) => (
    <Home {...defaultProps} {...props} />
  );

  it('should render Home root', () => {
    const wrapper = shallow(component({}));
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch fetchMaintainers action on mount', () => {
    const fetchMaintainers = spy();
    shallow(component({ fetchMaintainers }));

    expect(fetchMaintainers).to.have.been.calledWith(defaultProps.language);
  });

  it('should dispatch fetchMaintainers action on language change', () => {
    const fetchMaintainers = spy();
    const newLanguage = 'de';
    const wrapper = shallow(component({ fetchMaintainers, language: 'en' }));
    fetchMaintainers.resetHistory();
    wrapper.setProps({ language: newLanguage });

    expect(fetchMaintainers).to.have.been.calledWith(newLanguage);
  });
});
