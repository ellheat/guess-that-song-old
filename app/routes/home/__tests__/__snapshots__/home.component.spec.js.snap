// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Component should render Home root 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Homepage"
  />
  <styled.h1>
    <styled.i
      name="logo"
    />
    <FormattedMessage
      defaultMessage="Hello World!"
      id="home.welcome"
      values={Object {}}
    />
  </styled.h1>
  <styled.div>
    Environment: 
  </styled.div>
  <MaintainerList
    items={
      Immutable.List [
        Immutable.Map {
          "name": "name-1",
        },
        Immutable.Map {
          "name": "name-2",
        },
        Immutable.Map {
          "name": "name-3",
        },
      ]
    }
  />
</styled.div>
`;
